llvm/lib/Transforms/Scalar/StraightLineStrengthReduce.cpp

```diff
@@ -690,7 +694,8 @@ bool StraightLineStrengthReduce::runOnFunction(Function &F) {
   while (!Candidates.empty()) {
     const Candidate &C = Candidates.back();
     if (C.Basis != nullptr) {
-      rewriteCandidateWithBasis(C, *C.Basis);
+      if (DebugCounter::shouldExecute(StraightLineStrengthReduceCounter))
+        rewriteCandidateWithBasis(C, *C.Basis);
```

Comment:
Apologies for the last-minute review. 

After looking through other uses of debug counters, it seems like `shouldExecute` is usually called within the function that implements the transformation as opposed to guarding the call to the function that implements the transformation. 

i.e. 
```
if (shouldExecute())
  foo();
-->
void foo() {
  if (shouldExecute())
    return;
  ...
```

Can we make this change and then go forward with merging the PR?

Reply:
Good point, I have moved it to the top of `rewriteCandidateWithBasis` :)